tests/s3/copy_object/test_copy_obj_simultaneous_del.py

```diff
@@ -0,0 +1,232 @@
+            resp = self.s3_obj.create_bucket(bucket)
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Created %s bucket", bucket)
+        LOGGER.info("ENDED: Test setup.")
+
+    def teardown_method(self):
+        """Teardown for tests"""
+        LOGGER.info("STARTED: Test teardown.")
+        if os.path.exists(self.file_path):
+            os.remove(self.file_path)
+        LOGGER.info("Delete the objects and buckets created")
+        self.s3_obj.delete_multiple_buckets(self.buckets)
+        LOGGER.info("ENDED: Test teardown.")
+
+    @staticmethod
+    def copy_object_wrapper(src_bucket, src_obj, dest_bucket, dest_obj, exception=None):
+        """Copy object wrapper for multiprocessing"""
+        s3_obj = s3_test_lib.S3TestLib()
+        try:
+            return s3_obj.copy_object(src_bucket, src_obj, dest_bucket, dest_obj)
+        except CTException as err:
+            LOGGER.info("Exception in copy object %s", err)
+            if exception:
+                return exception in err.message, f"Expected {exception} Received {err}"
+            return False, f"Unexpected exception {err}"
+
+    def create_put_object(self, bucket, obj, size, base='1M'):
+        """Create and Put object of (size*1M) size and assert"""
+        resp = system_utils.create_file(fpath=self.file_path, count=size, b_size=base)
+        assert_utils.assert_true(resp[0], resp[1])
+        return self.s3_obj.put_object(bucket, obj, self.file_path)
+
+    @staticmethod
+    def delete_object_wrapper(bucket, obj):
+        """ Delete object wrapper for multiprocessing"""
+        s3_obj = s3_test_lib.S3TestLib()
+        return s3_obj.delete_object(bucket, obj)
+
+    def create_put_parallel_copy_and_delete_object(self, copy_args, delete_args, is_mpu=False):
+        """Create and uplaod object to bucket and then Parallel copy object and put object with
+        given arguments"""
+        if is_mpu:
+            resp = system_utils.create_file(fpath=copy_args[7], count=copy_args[6], b_size="1M")
+            assert_utils.assert_true(resp[0], resp[1])
+            resp = self.s3mp_obj.simple_multipart_upload(copy_args[0], copy_args[1], copy_args[5],
+                                                         copy_args[7], copy_args[6])
+        else:
+            self.create_put_object(copy_args[0], copy_args[1], copy_args[5])
+        LOGGER.info("Parallely, \n1. From %s copy %s to %s as %s\n2. Delete %s from %s",
+                    copy_args[0], copy_args[1], copy_args[2], copy_args[3], delete_args[1],
+                    delete_args[1])
+        with multiprocessing.Pool(processes=2) as pool:
+            process1 = pool.apply_async(self.copy_object_wrapper, args=copy_args[:5])
+            process2 = pool.apply_async(self.delete_object_wrapper, args=delete_args)
+            assert_utils.assert_true(process1.get()[0], process1.get()[1])
+            assert_utils.assert_true(process2.get()[0], process2.get()[1])
```

Comment:
Need to add verify either NoSuchKey error on copy object or both succeed

Reply:
This is being taken care by copy_object_wrapper method